### PR TITLE
Created WETH wrapper

### PR DESCRIPTION
**Pull request by nivgoldstein**

### CODE REVIEW

File: wrap.py [hunk 1]
```diff
@@ -1,29 +1,68 @@
...
+    def __init__(self, private_key: str, testnet=True):
+        self._tx_timeout = 150
+        self.private_key = private_key
+        self.account = Account.privateKeyToAccount(self.private_key)
+
+        if testnet:
+            self.web3 = Web3(Web3.HTTPProvider(INFURA_RINKEBY_HTTP))
+            contract_address = RINKEBY_WETH_CONTRACT
+        else:
+            self.web3 = Web3(Web3.WebsocketProvider(endpoint_uri=INFURA_MAINNET_WEBSOCKET, websocket_timeout=1800))
```
Comment (oleg-efrontier) [inline wrap.py:25]: Is there a  need for websockets ? Are we subscribing to events or use any other feature of websockets provider?

File: wrap.py [hunk 2]
```diff
@@ -1,29 +1,68 @@
...
-        result = rej.json()
-        self.currencies = {x['symbol']: x for x in result['data']}
-        return self.currencies
+    def __init__(self, private_key: str, testnet=True):
+        self._tx_timeout = 150
+        self.private_key = private_key
+        self.account = Account.privateKeyToAccount(self.private_key)
+
+        if testnet:
+            self.web3 = Web3(Web3.HTTPProvider(INFURA_RINKEBY_HTTP))
```
Comment (oleg-efrontier) [inline wrap.py:22]: I would suggest to place network-dependent constants under one config dictionary:
```
   NETWORKS = {
        'mainnet': { 'http_provider': xxx1, 'weth9_sc_address': yyy1 },
        'ropsten':{ 'http_provider': xxx2, 'weth9_sc_address': yyy2 },
        'rinkeby': { 'http_provider': xxx3, 'weth9_sc_address': yyy3 },
    }
```

File: wrap.py [hunk 3]
```diff
@@ -1,29 +1,68 @@
...
...'outputs': [], 'payable': True, 'stateMutability': 'payable', 'type': 'function'}, {'constant': True, 'inputs': [{'name': '', 'type': 'address'}, {'name': '', 'type': 'address'}], 'name': 'allowance', 'outputs': [{'name': '', 'type': 'uint256'}], 'payable': False, 'stateMutability': 'view', 'type': 'function'}, {'payable': True, 'stateMutability': 'payable', 'type': 'fallback'}, {'anonymous': False, 'inputs': [{'indexed': True, 'name': 'src', 'type': 'address'}, {'indexed': True, 'name': 'guy', 'type': 'address'}, {'indexed': False, 'name': 'wad', 'type': 'uint256'}], 'name': 'Approval', 'type': 'event'}, {'anonymous': False, 'inputs': [{'indexed': True, 'name': 'src', 'type': 'address'}, {'indexed': True, 'name': 'dst', 'type': 'address'}, {'indexed': False, 'name': 'wad', 'type': 'uint256'}], 'name': 'Transfer', 'type': 'event'}, {'anonymous': False, 'inputs': [{'indexed': True, 'name': 'dst', 'type': 'address'}, {'indexed': False, 'name': 'wad', 'type': 'uint256'}], 'name': 'Deposit', 'type': 'event'}, {'anonymous': False, 'inputs': [{'indexed': True, 'name': 'src', 'type': 'address'}, {'indexed': False, 'name': 'wad', 'type': 'uint256'}], 'name': 'Withdrawal', 'type': 'event'}]
```
Comment (oleg-efrontier) [inline wrap.py:5]: You need either file or JSON , not both...

File: wrap.py (hunk 1 above)
Comment (nivgoldstein) [reply wrap.py:25]: According to the docs it's preferable to use websocket provider when possible

Comment (nivgoldstein) [reply wrap.py:25]: If anything, I'll make rinkeby use websocket as well

Comment (oleg-efrontier) [reply wrap.py:25]: Just make it consistent.

File: wrap.py (hunk 2 above)
Comment (nivgoldstein) [reply wrap.py]: Done

File: wrap.py (hunk 3 above)
Comment (nivgoldstein) [reply wrap.py:3]: Done

Comment (nivgoldstein) [reply wrap.py:3]: Done

File: wrap.py [hunk 4]
```diff
@@ -1,36 +1,38 @@
...
+            'contract_address': '0xc778417e063141139fce010982780140aa0cd5ab'
+        },
+        'mainnet': {
+            'ws_endpoint_uri': 'wss://mainnet.infura.io/ws/v3/34e00ade3f6c4facbf519c9c3670f025',
+            'contract_address': '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+        }
+    }
 
-class eth_wrapper:
     def __init__(self, private_key: str, testnet=True):
```
Comment (oleg-efrontier) [inline wrap.py:18]: Use
     def __init__(self, private_key: str, testnet='rinkeby'):
and you won't need IF/else.

Comment (nivgoldstein) [reply wrap.py]: Done

**Pull request merged**